.github/workflows/ci.yaml

```diff
@@ -8,177 +8,303 @@ on:
+
+      - name: Build vscode
+        if: steps.cache-vscode.outputs.cache-hit != 'true'
+        run: yarn build:vscode
+
+      # The release package does not contain any native modules
+      # and is neutral to architecture/os/libc version.
+      - name: Create release package
+        run: yarn release
+        if: success()
+
+      # https://github.com/actions/upload-artifact/issues/38
+      - name: Compress release package
+        run: tar -czf package.tar.gz release
+
       - name: Upload npm package artifact
         uses: actions/upload-artifact@v2
         with:
           name: npm-package
-          path: ./release-npm-package
+          path: ./package.tar.gz
+
+  # We use Ubuntu 16.04 here, so that our build is more compatible
+  # with older libc versions. We used to (Q1'20) use CentOS 7 here,
+  # but it has a full update EOL of Q4'20 and a 'critical security'
+  # update EOL of 2024. We're dropping our full support 3 quarters
+  # early, but this gets us CI speed benefits on GH.
+
+  # TODO: cache building yarn --production
+  # possibly 2m30s of savings(?)
+  # this requires refactoring our release scripts
+
+  # TODO: cross-compile on arm64
+  package:
+    name: x86-64 builds
+    needs: build
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-16.04, macos-latest]
+      fail-fast: false
 
-  linux-amd64:
-    needs: release
-    runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - name: Download npm package
-        uses: actions/download-artifact@v2
-        with:
-          name: npm-package
-          path: ./release-npm-package
-      - name: Run ./ci/steps/release-packages.sh
-        uses: ./ci/images/centos7
-        with:
-          args: ./ci/steps/release-packages.sh
-      - name: Upload release artifacts
-        uses: actions/upload-artifact@v2
+
+      - name: Install node.js v12
+        uses: actions/setup-node@v2
         with:
-          name: release-packages
-          path: ./release-packages
+          node-version: "12"
+
+      - name: Install nfpm
+        run: |
+          curl -sfL https://install.goreleaser.com/github.com/goreleaser/nfpm.sh | sh -s -- -b ~/.local/bin v2.3.1
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
 
-  linux-arm64:
-    needs: release
-    runs-on: ubuntu-arm64-latest
-    steps:
-      - uses: actions/checkout@v2
       - name: Download npm package
         uses: actions/download-artifact@v2
         with:
           name: npm-package
-          path: ./release-npm-package
-      - name: Run ./ci/steps/release-packages.sh
-        uses: ./ci/images/centos7
-        with:
-          args: ./ci/steps/release-packages.sh
+
+      - name: Decompress npm package
+        run: tar -xzf package.tar.gz
+
+      - name: Build standalone release
+        run: yarn release:standalone
+
+      - name: Sanity test standalone release
+        run: yarn test:standalone-release
```

Comment:
I wonder if this should be grouped with the testing workflow

Reply:
I'm not sure if we should move this to the e2e workflow - this test essentially downloads and installs a vscode theme extension, just as a very quick test to see if `code-server` can run correctly. What do you think we should do?

Reply:
I'll lean towards your judgement! If you think it's fine here, let's leave it. We can always move it later if needed!